### PR TITLE
Show warning when unsupported Devfile features used

### PR DESCRIPTION
**Pull request by akurinnoy**
### What does this PR do?

This PR show warnings that may happen during workspace creation when the devfile uses unsupported features.

Screenshot 2023-03-03 at 18 44 54

### What issues does this PR fix or reference?

resolves https://github.com/eclipse/che/issues/21873

### Is it tested? How?

Try creating a workspace using the following devfile: 

```yaml
schemaVersion: 2.1.0
metadata:
  generateName: unsupported-feature-
components:
  - name: web-terminal
    container:
      image: quay.io/wto/web-terminal-tooling:next
      memoryLimit: 512Mi
      mountSources: true
      command:
       - "tail"
       - "-f"
       - "/dev/null"
      dedicatedPod: true # This feature is unsupported
```

### CODE REVIEW

**Review by ibuziuk — APPROVED** [review]

**Comment by ibuziuk** [comment]
@amisevsk @AObuchow folks, could you please review
I'm a bit concerned about the warning code, probably we should not expose it on the UD level

**Review by amisevsk — APPROVED** [review]
LGTM in general. I would ignore the HTTP status code if possible as it could be confusing to users (e.g. Kubernetes uses 299 for webhook warnings, but it's not necessarily relevant to the UI).

Worth noting: I haven't reviewed the code itself very thoroughly.

**Review by AObuchow — APPROVED** [review]
LGTM, though I agree with @ibuziuk and @amisevsk that I don't think the HTTP status code should be shown in the UI (if possible) as it doesn't seem very useful to the user.

File: packages/dashboard-frontend/src/services/workspace-client/devworkspace/__tests__/devWorkspaceClient.create.spec.ts [hunk 1]
```diff
@@ -82,60 +89,24 @@ describe('DevWorkspace client, create', () => {
...
-        .spyOn(DwApi, 'patchWorkspace')
-        .mockResolvedValueOnce(testDevWorkspace);
-      const workspace = new DevWorkspaceBuilder()
-        .withMetadata({
-          namespace,
-          name: 'wksp-name-cvbn',
-          uid: 'uid-asdfgh12345',
-        })
-        .build();
-      spyCreateWorkspace = jest.spyOn(DwApi, 'createWorkspace').mockResolvedValueOnce(workspace);
```
Comment (olexii4) [inline packages/dashboard-frontend/src/services/workspace-client/devworkspace/__tests__/devWorkspaceClient.create.spec.ts:96]: Could you remove unneeded imports('DwApi' and 'DevWorkspaceBuilder') too?

**Review by olexii4 — APPROVED** [review]

**Pull request merged**